conans/model/settings.py

```diff
@@ -97,11 +99,18 @@ def __eq__(self, other):
         other = str(other)
         if self._not_any() and other not in self.values_range:
             raise ConanException(bad_value_msg(self._name, other, self.values_range))
-        return other == self.__str__()
+        if self._name != "settings.compiler.cppstd":
+            return other == self.__str__()
+        return int(self._value.replace("gnu", "")) == int(str(other).replace("gnu", ""))
```

Comment:
Yes, I think this is not a good approach. This will be broken (a regression, with an ugly stack trace) for anyone who added custom settings like ``C++2a``